docs/aws/audit/lambdamonitoring/rules/admin_privilege_remediation.mdx

```diff
@@ -1,6 +1,270 @@
+functions = lambda_client.list_functions()
+
+for function in functions['Functions']:
+    function_name = function['FunctionName']
+    role = function['Role']
+    role_name = role.split('/')[-1]
+    
+    enforce_least_privilege(function_name, role_name)
+```
+
+By following these steps, you can ensure that your AWS Lambda functions do not have administrative permissions and adhere to the principle of least privilege.
+</Accordion>
+
+</AccordionGroup>
+</Tab>
+<Tab title='Cause'>
+### Check Cause
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Sign in to the AWS Management Console.
+2. Navigate to the IAM dashboard by selecting "Services" and then "IAM".
+3. In the IAM dashboard, select "Policies" from the left-hand navigation menu.
+4. In the Policies page, use the Filter to find policies that are attached to your Lambda functions. You can do this by typing "lambda" into the Filter and pressing enter.
+5. Once you have the list of policies attached to your Lambda functions, you can check each policy to see if it has administrative permissions. To do this, click on the policy name to open its details page. In the "Permissions" tab, look for permissions that include "AdministratorAccess". If such permissions are found, then the Lambda function has administrative permissions, which is a misconfiguration.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to list all the Lambda functions in your AWS account. You can do this by using the AWS CLI command `aws lambda list-functions`. This command will return a list of all the Lambda functions in your AWS account.
+
+2. Once you have the list of Lambda functions, you can check the permissions attached to each function. You can do this by using the AWS CLI command `aws lambda get-policy --function-name <function-name>`. Replace `<function-name>` with the name of the Lambda function you want to check. This command will return the policy attached to the specified Lambda function.
+
+3. Now, you need to parse the output of the `get-policy` command to check if the Lambda function has administrative permissions. You can do this by looking for the `"Effect": "Allow"` and `"Action": "lambda:*"` fields in the policy. If these fields are present, it means that the Lambda function has administrative permissions.
+
+4. Repeat steps 2 and 3 for each Lambda function in your AWS account. If any function has administrative permissions, it is a misconfiguration.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Import the necessary libraries: You will need the boto3 library, which allows Python developers to write software that makes use of services like Amazon S3, Amazon EC2, etc.
+
+```python
+import boto3
+```
+
+2. Create a session using your AWS credentials. You can also use the AWS CLI's default configuration.
+
+```python
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>